src/density_methods.jl

```diff
@@ -140,18 +140,18 @@ end
 # Perform an atomic density superposition. The density is constructed in reciprocal space
 # using the provided atomic form-factors and coefficients and applying an inverse Fourier
 # transform to yield the real-space density.
-function atomic_density_superposition(basis::PlaneWaveBasis{T},
-                                      form_factors::IdDict{Tuple{Int,T},T};
+function atomic_density_superposition(basis::PlaneWaveBasis{T,VT},
+                                      form_factors::Dict{Tuple{Int,VT},T};
                                       coefficients=ones(T, length(basis.model.atoms))
-                                      ) where {T}
+                                      ) where {T,VT}
     model = basis.model
     G_cart = to_cpu(G_vectors_cart(basis))
     ρ_cpu = map(enumerate(to_cpu(G_vectors(basis)))) do (iG, G)
-        Gnorm = norm(G_cart[iG])
+        G_val = ForwardDiff.value(norm(G_cart[iG]))
```

Comment:
Add comment? This feels very dangerous to just discard duals like that but I haven't looked at it too closely

Reply:
Actually this does not work in case you have duals in the pseudo parameters -> undone for safety.